MediaGalleryUi/view/adminhtml/web/js/image/image-edit.js

```diff
@@ -20,11 +22,102 @@ define([
             saveDetailsUrl: '/media_gallery/image/saveDetails',
             images: [],
             image: null,
+            keywordOptions: [],
+            selectedKeywords: [],
+            newKeyword: '',
+            newKeywordSelector: '#keyword',
             modules: {
-                mediaGridMessages: '${ $.mediaGridMessages }'
+                mediaGridMessages: '${ $.mediaGridMessages }',
+                keywordsSelect: '${ $.name }_keywords'
+            },
+            viewConfig: [
+                {
+                    component: 'Magento_Ui/js/form/element/ui-select',
+                    name: '${ $.name }_keywords'
+                }
+            ],
+            exports: {
+                keywordOptions: '${ $.name }_keywords:options'
+            },
+            links: {
+                selectedKeywords: '${ $.name }_keywords:value'
```

Comment:
I believe we should create a separate component like `ui-keywords` that extends `ui-select`, the way it is at the moment we have `image-edit` component changing linked properties and as consequence changing directly the `ui-select` state. By having separate component the component would have its own template and its own methods changing its own properties. What do you think?

Reply:
That's not actually keywords ui-select responsibility to change it's options/value. We may introduce a separate component for add new keyword field and button. However, in this case image-edit will still needs to be a proxy between ui-select and add-keyword components, so in my opinion that might increase the complexity. I'd postpone extracting additional components from `image-edit`

Reply:
As for adding the add keyword field to ui-select option - I wouldn't like to rewrite the ui-select component and template as that is a potential for conflicts on upgrades